playwright-tests/tests/proposals.spec.js

```diff
@@ -455,4 +455,45 @@ test.describe("Wallet is connected", () => {
 
     await pauseIfVideoRecording(page);
   });
+
+  test("should filter proposals by categories", async ({ page }) => {
+    test.setTimeout(120000);
+    await page.goto("/devhub.near/widget/app?page=proposals");
+    await page.getByRole("button", { name: "Category" }).click();
+    await page.getByRole("list").getByText("DevDAO Operations").click();
+    await expect(
+      page.getByRole("button", { name: "Category : DevDAO Operations" })
+    ).toBeVisible();
+    test.setTimeout(4000);
+  });
+
+  test("should filter proposals by timeline", async ({ page }) => {
+    test.setTimeout(120000);
+    await page.goto("/devhub.near/widget/app?page=proposals");
+    await page.getByRole("button", { name: "Stage" }).click();
+    await page.getByRole("list").getByText("Funded").click();
+    await expect(
+      page.getByRole("button", { name: "Stage : Funded" })
+    ).toBeVisible();
+    test.setTimeout(4000);
+  });
+
+  test("should filter proposals by author", async ({ page }) => {
+    test.setTimeout(120000);
+    await page.goto("/devhub.near/widget/app?page=proposals");
+    await page.getByRole("button", { name: "Author" }).click();
+    await page.getByRole("list").getByText("megha19.near").click();
+    await expect(
+      page.getByRole("button", { name: "Author : megha19.near" })
+    ).toBeVisible();
+    test.setTimeout(4000);
```

Comment:
why setting timeout here?

Reply:
to wait for new filtered data to be fetched.

Reply:
`test.setTImeout` is for setting how long the test is allowed to run ( default is 30 seconds ). If you need to wait for something to appear on the page, you can normally just use `await expect(await page.locator(".some-class-name")).toHaveText("blabla")`, like you can see here: https://github.com/NEAR-DevHub/neardevhub-bos/pull/802/files#diff-1765f31365d3e8010bc399faa0e2b805b801dbf1cb84c5abfe7e06184f57fa28R329

The default setting is that it waits for up to 5  seconds, but you can also add more as a setting to the `page.locator` method.